src/Estimators/src/UkfState.cpp

```diff
@@ -0,0 +1,543 @@
+    m_pimpl->isFinalized = false;
+
+    m_pimpl->stateSize = 0;
+
+    // finalize all the dynamics
+    for (int indexDyn1 = 0; indexDyn1 < m_pimpl->dynamicsList.size(); indexDyn1++)
+    {
+        if (!m_pimpl->dynamicsList[indexDyn1].second->finalize(handler))
+        {
+            log()->error("{} Error while finalizing the dynamics named {}",
+                         logPrefix,
+                         m_pimpl->dynamicsList[indexDyn1].first);
+            return false;
+        }
+
+        m_pimpl->stateSize += m_pimpl->dynamicsList[indexDyn1].second->size();
+    }
+
+    // Set value of process covariance matrix
+    m_pimpl->covarianceQ.resize(m_pimpl->stateSize, m_pimpl->stateSize);
+    m_pimpl->covarianceQ.setZero();
+
+    m_pimpl->initialCovariance.resize(m_pimpl->stateSize, m_pimpl->stateSize);
+    m_pimpl->initialCovariance.setZero();
+
+    for (int indexDyn2 = 0; indexDyn2 < m_pimpl->dynamicsList.size(); indexDyn2++)
+    {
+        m_pimpl->covarianceQ
+            .block(handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).offset,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).offset,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).size,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).size)
+            = m_pimpl->dynamicsList[indexDyn2].second->getCovariance().asDiagonal();
+
+        m_pimpl->initialCovariance
+            .block(handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).offset,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).offset,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).size,
+                   handler.getVariable(m_pimpl->dynamicsList[indexDyn2].first).size)
+            = m_pimpl->dynamicsList[indexDyn2].second->getInitialStateCovariance().asDiagonal();
+    }
+
+    m_pimpl->jointVelocityState.resize(m_pimpl->kinDynFullModel->model().getNrOfDOFs());
+    m_pimpl->jointAccelerationState.resize(m_pimpl->kinDynFullModel->model().getNrOfDOFs());
+
+    for (int idx = 0; idx < m_pimpl->subModelList.size(); idx++)
+    {
+        m_pimpl->subModelJointVel.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
+        m_pimpl->subModelJointAcc.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
+        m_pimpl->subModelJointMotorTorque.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
+        m_pimpl->subModelFrictionTorque.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
+        m_pimpl->totalTorqueFromContacts.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
+        m_pimpl->torqueFromContact.emplace_back(
+            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
```

Comment:
`emplace_back` should automatically call the constructor. So in theory your code does the following
```cpp
m_pimpl->subModelJointVel.emplace_back(
            Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs()));
```
1. create a lvalue `Eigen::VectorXd(m_pimpl->subModelList[idx].getModel().getNrOfDOFs())`
2. copy or moved depending on what is defined in eigen into a new `Eigen::VectorXd` (I'm afraid that the copy is called)
3. Delete the `lvalue`

In theory, what you can do is 
```cpp
 m_pimpl->subModelJointVel.emplace_back(m_pimpl->subModelList[idx].getModel().getNrOfDOFs());
```

In this case the object is automatically build